Source/WebCore/page/scrolling/ScrollAnchoringController.cpp

```diff
@@ -174,7 +174,7 @@ CandidateExaminationResult ScrollAnchoringController::examineCandidate(Element&
         // TODO: figure out how to get scrollable area for renderer to check if it is maintaining scroll anchor
         if (renderer->style().overflowAnchor() == OverflowAnchor::None || renderer->isStickilyPositioned() || renderer->isFixedPositioned() || renderer->isPseudoElement() || renderer->isAnonymousBlock())
             return CandidateExaminationResult::Exclude;
-        if (&element == document->bodyOrFrameset() || is<HTMLHtmlElement>(&element) || (renderer->isInline() && !renderer->isAtomicInlineLevelBox()))
+        if (&element == document->bodyOrFrameset() || is<HTMLHtmlElement>(&element) || (is<RenderBox>(renderer) && renderer->isInline() && !renderer->isAtomicInlineLevelBox()))
```

Comment:
I don't think this change reflects what the commit message says:
"We should only check if a renderer is a non-atomic inline box for boxes."
e.g. RenderInline represents a non-atomic inline box but it is not a type of RenderBox. Unless "for boxes" literally means (Render)Boxes but is that what the spec says?

Reply:
The spec has this as a criteria for being chosen as the anchor element 
> C is an element that is not a non-atomic inline
where it uses https://www.w3.org/TR/css-display-3/#atomic-inline as its definition of an atomic inline. So I think you're right that this pr isn't accurate to the spec, but I don't think this check should apply to the text nodes in the relevant tests based on the atomic inline definition.

Reply:
correct (and I didn't meant to imply text should be involved at all) text is not an inline level element. What I was trying to refer was actual inline boxes (RenderInline objects).

Reply:
Ok I see. So then a sufficient check would be something like this: 
`is<RenderInline>(renderer) &&  !renderer->isAtomicInlineLevelBox()`.